hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java

```diff
@@ -441,6 +441,19 @@ public List<BlockData> listBlock(Container container, long startLocalID, int
     }
   }
 
+  @Override
+  public boolean blockExists(Container container, BlockID blockID) throws IOException {
+    KeyValueContainerData containerData = (KeyValueContainerData) container
+        .getContainerData();
+    try (DBHandle db = BlockUtils.getDB(containerData, config)) {
+      // This is a post condition that acts as a hint to the user.
+      // Should never fail.
```

Comment:
This comment is useless. Isn't it the reason why the precondition check is here?

Reply:
I agree, but it's already copy/pasted all over `BlockManagerImpl` 😄